### PR TITLE
Change to new C++ based addon interface style

### PR DESCRIPTION
**Pull request by AlwinEsch**
Related to Kodi request https://github.com/xbmc/xbmc/pull/12318

### CODE REVIEW

File: src/RawPicture.cpp [hunk 1]
```diff
@@ -18,64 +18,63 @@
...
 
-class RawPicture
+class RawPicture : public kodi::addon::CInstanceImageDecoder
 {
 public:
-  RawPicture()
+  RawPicture(KODI_HANDLE instance)
+    : CInstanceImageDecoder(instance),
+      m_raw_data(libraw_init(0)),
+      m_thumbnailbuffer(nullptr)
```
Comment (notspiff) [inline src/RawPicture.cpp]: is this used?

Comment (AlwinEsch) [inline src/RawPicture.cpp]: Let me look.

Comment (AlwinEsch) [inline src/RawPicture.cpp]: Have looked not used :), there is also a function call with `libraw_unpack_thumb` for where it is needed?

Comment (notspiff) [inline src/RawPicture.cpp]: initial version of image decoder api had thumb handling, but it was moved completely to core. so it's a leftover.

**Comment by AlwinEsch** [comment]
Is updated.

**Pull request merged**